vm/map.c

```diff
@@ -775,11 +775,29 @@ void vm_mapDestroy(process_t *p, vm_map_t *map)
 
 	proc_lockDone(&map->lock);
 #else
-	proc_lockSet(&map->lock);
-	while ((e = p->entries) != NULL) {
-		_map_remove(map, e);
-		map_free(e);
+	map_entry_t *next;
+
+	proc_lockSet2(&map->lock, &p->lock);
+	if (p->entries != NULL) {
+		e = p->entries;
+		do {
```

Comment:
`if` maybe removed

```c
e = p->entries;
while ((e != NULL) && (e != p->entries))
```

Reply:
Nope, if e == p-> entries we won't enter such while

Reply:
Ow, right